### PR TITLE
Added Authenticate

### PR DESCRIPTION
**Pull request by haarsh24**
1. Added login authentication
2. Added Signup authentication

### CODE REVIEW

**Review by yashpatel1797 — APPROVED** [review]
the file structure is good.
design is also consistent.
try to remove comments from the code.

File: src/contexts/authContext.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { createContext,useContext ,useReducer} from "react";   
+import { authReducer } from "../reducer/authReducer";
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+    // const { firstName } = JSON.parse(localStorage.getItem('userData')) || { firstName: "" }
+    // let tokenDetails = localStorage.getItem("token") || { token: '' }
```
Comment (xii-vi) [inline src/contexts/authContext.js:7]: _Remove the unused code_

File: src/contexts/authContext.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const [authState, authDispatch] = useReducer(authReducer, {
+    isLogin:false,
+    token:"",
+    userDetails:""
+});
+// isLogin: (tokenDetails !== ''),
```
Comment (xii-vi) [inline src/contexts/authContext.js:13]: _Same here_

**Review by xii-vi — APPROVED** [review]
_Go through the comments before merging._

**Pull request merged**